### PR TITLE
Update for BaseClassDefinitions TFS:9444250

### PR DESCRIPTION
**Pull request by anmenaga**

### CODE REVIEW

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 1]
```diff
@@ -15,7 +15,10 @@ Describe "Test suite for Microsoft.PowerShell.ODataUtils module" -Tags "BVT" {
 
         BeforeAll {
             $scriptToDotSource = Join-Path $ModuleBase 'Microsoft.PowerShell.ODataUtilsHelper.ps1'
+            $eap = $ErrorActionPreference
+            $ErrorActionPreference = 'SilentlyContinue' # ignore "New-Variable : A variable with name 'BaseClassDefinitions' already exists."
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: This set of code seems repeated enough to be a function

Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

**Comment by anmenaga** [comment]
This module-specific repo has latest version of the module and, yes, paths are slightly different from a 'old' version in PowerShell/PowerShell.
As soon as this PR approved, I'll make corresponding changes in RS2 branch and PowerShell/PowerShell.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 2]
```diff
@@ -9,13 +9,23 @@ Describe "Test suite for Microsoft.PowerShell.ODataUtils module" -Tags "BVT" {
...
+        return $eap
     }
 
     Context "OData validation test cases" {
 
         BeforeAll {
-            $scriptToDotSource = Join-Path $ModuleBase 'Microsoft.PowerShell.ODataUtilsHelper.ps1'
-            . $scriptToDotSource
+            $eap = SaveAndSetErrorActionPreference
+            . $ODataUtilsHelperPath
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: the pattern you're using is:

1. save current erroractionpref
2. dotsource a script
3. restore erroractionpref

I think you should change the SaveAndSetErrorActionPreference function to a DotSourceScriptAndRestoreErrorActionPreference function.

Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Removed function completely and moved dotsourcing of Helper.ps1 to beginning of test script.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 3]
```diff
@@ -4,18 +4,17 @@
...
+$ModuleBase = Join-Path (Join-Path $ModuleBase 'src') 'ModuleGeneration'
 
-    BeforeAll {
-        $ModuleBase = Split-Path $script:TestSourceRoot
-        $ModuleBase = Join-Path (Join-Path $ModuleBase 'src') 'ModuleGeneration'
-    }
+$ODataUtilsHelperPath = Join-Path $ModuleBase 'Microsoft.PowerShell.ODataUtilsHelper.ps1'
+. $ODataUtilsHelperPath
+
+Describe "Test suite for Microsoft.PowerShell.ODataUtils module" -Tags "BVT" {
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: We should use the standardized CI tag from PS/PS instead of BVT

Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

**Pull request merged**